chapters/annotations.tex

```diff
@@ -8,6 +8,12 @@ \chapter{Annotations}\label{annotations}
 (For replaceable class declarations with a \lstinline[language=grammar]!constraining-clause! also refer to \cref{constraining-clause-annotations}.)
 The specification in this document defines the semantic meaning if a tool implements any of these annotations.
 
+\section{How to Interpret Annotation Definitions}\label{how-to-interpret-annotation-definitions}
+If the annotation itself is described by a \lstinline!record! \emph{class} the annotation can be seen as a modifier for a \lstinline!record! \emph{component} with the same name.
+A default value for a record member will be used if there is no modifier for that record member.
+If the annotation is declared as an \lstinline!/*evaluable*/ parameter! the corresponding modifier must be an evaluable parameter.
```

Comment:
I don't think we should require it to be a parameter, right?
```suggestion
If the annotation is declared as an \lstinline!/*evaluable*/ parameter! the corresponding modifier must be evaluable.
```

Reply:
However, first we should explain the general idea of specifying variability:
```suggestion
When an annotation is defined with a component variability prefix (\cref{component-variability-prefixes-discrete-parameter-constant}), this restricts the allowed variability of corresponding annotation modifiers analogously to the rules in \cref{variability-of-expressions}.
If the annotation is declared as an \lstinline!/*evaluable*/ parameter! the corresponding modifier is further restricted to be evaluable.
```

Reply:
So far there is just one instance of parameter and it is evaluable. Has there been an agreement on which annotations are allowed to be parameters and not evaluable? Given that the annotations are not supposed to store extra information, I would be very suspicious of any annotation that uses a parameter but that parameter can only solved in initial equations. I only know of DynamicSelect that has simulation time dependency. But then it allows variables, so even there the case of non-evaluable parameter is not applicable. All of this is to say, do we really need the case for parameters that are not evaluable?

Reply:
I think the plan is to start with a conservative interpretation of the current state, and then have later discussions in other PRs about what variability restrictions we can lift.

I think the case for parameters that are not evaluable is at least good for completeness, and even if we didn't have that case I would still have preferred to say `/*evaluable*/ parameter` for the evaluable ones in order to make clear that there's a difference compared to the usual meaning of declaring something as just `parameter` (for which there is not requirement of being evaluable).

Reply:
Actually, there's at least one good candidate for pure `parameter` variability, namely `experiment.StopTime`.  The `StartTime` is more complicated due to its relation to `time` during initialization, but it wouldn't be impossible to handle in the form of a semantic constraint.

Reply:
However, even if possible it does not seem urgent as we can already handle such models as follows:
```
  parameter Real myStopTime=...;
equation
  when time>=myStopTime then
    terminate("Planned end");
  end when;
```

Reply:
I didn't say it was urgent. Anyway, using `terminate` isn't quite the same as setting `StopTime`, as it requires specifying some insanely big `StopTime` that is guaranteed to exceed `myStopTime`.  However, doing so might have adverse impact on initial guesses for solver step size, so it would be better to directly control `StopTime` using the parameter.